web/html/src/manager/state/recurring-states-edit.js

```diff
@@ -0,0 +1,137 @@
+/* eslint-disable */
+'use strict';
+
+const React = require("react");
+const ReactDOM = require("react-dom");
+const {RecurringEventPicker} = require("components/recurring-event-picker");
+const {DisplayHighstate} = require("./display-highstate");
+const Button = require("components/buttons").Button;
+const AsyncButton = require("components/buttons").AsyncButton;
+const {Toggler} = require("components/toggler");
+const { InnerPanel } = require("components/panels/InnerPanel");
+
+class RecurringStatesEdit extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            minions: minions,
+            active: true
+        };
+
+        if(this.isEdit()) {
+            this.setSchedule(this.props.schedule);
+        } else {
+            this.getTargetType();
+        }
+    }
+
+    setSchedule = (schedule) => {
+        Object.assign(this.state, schedule);
+    };
+
+    getTargetType = () => {
+        const search = window.location.search;
+        if (search.match("\\?sid")) {
+            Object.assign(this.state, {
+                targetType: "MINION", // todo create a human readable representations
+                targetId: minions[0].id
+            });
+        } else if (search.match("\\?sgid")) {
+            Object.assign(this.state, {
+                targetType: "GROUP",
+                targetId: groupId
+            });
+        } else {
+            Object.assign(this.state, {
+                targetType: "ORG",
+                targetId: orgId
+            });
+        }
+    };
+
+    isEdit = () => {
+        return !!this.props.schedule;
```

Comment:
I am worried about this obscure magic :smile: 

I understand the reason of the logic, but since it is already extracted in a separate method, I'd really explain more the logic in a different way. I can't imagine debugging or understanding this from a new pair of eyes.